### PR TITLE
Rewrite data generation pipeline without Shrike

### PR DESCRIPTION
**Pull request by jfomhover**
This PR is an attempt at rewriting the data generation pipeline to use purely ComponentSDK constructs, no shrike, while still remaining compatible with the config files in yaml.

In order to achieve that, here are the multiple changes:
- rewriting pipeline without AMLPipelineHelper class: reduces complexity for our pipelines (just a couple modules), and makes it more readable, also offers greater control for creating workarounds
- rewriting the `AMLPipelineHelper.main()` method: the heart of this PR, it consisted in taking bits of the shrike main() function and keep only what's necessary in our use case (validate, submit). The helper function has been segmented into 3 functions (parse_pipeline_config, azureml_connect, pipeline_submit) so users have to write some main block, but it offers more flexibility.
- copying some of the dataclasses from shrike into our code base (aml connection settings, compute settings etc), we used this opportunity to change the compute settings to be very minimal (linux cpu, linux gpu, windows cpu)

### CODE REVIEW

File: src/common/pipelines.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    """
+    return shrike_azureml_connect(
+        aml_subscription_id=config.aml.subscription_id,
+        aml_resource_group=config.aml.resource_group,
+        aml_workspace_name=config.aml.workspace_name,
+        aml_auth=config.aml.auth,
+        aml_tenant=config.aml.tenant
+    )
+
+def pipeline_submit(workspace,
```
Comment (dkmiller) [inline src/common/pipelines.py:143]: Suggestion: add type hints for the parameters when possible.

File: pipelines/azureml/pipelines/data_generation.py [hunk 2]
```diff
@@ -1,195 +1,161 @@
 """
-Runs LightGBM using distributed (mpi) training.
+Generates synthetic data with multiple parameters.
+
+See config file /pipelines/azureml/conf/experiments/data-generation.yaml
 
 to execute:
-> python pipelines/data_generation.py --config-dir ./conf --config-name experiments/data-generation run.submit=True
+> python pipelines/azureml/pipelines/data_generation.py --exp-config ./pipelines/azureml/conf/experiments/data-generation.yaml +run.submit=True
```
[the remark this answers is not in the document]
Comment (jfomhover) [reply pipelines/azureml/pipelines/data_generation.py]: yes! done

File: pipelines/azureml/conf/experiments/data-generation.yaml [hunk 3]
```diff
@@ -1,35 +1,24 @@
 # This experiment generates multiple synthetic datasets for regression
 # with varying number of features
 #
-# to execute, run from /pipelines/azureml/
-# > python pipelines/data_generation.py --config-dir ./conf --config-name experiments/data-generation run.submit=True
+# to execute:
+# > python pipelines/azureml/pipelines/data_generation.py --exp-config ./pipelines/azureml/conf/experiments/data-generation.yaml
```
[the remark this answers is not in the document]
Comment (jfomhover) [reply pipelines/azureml/conf/experiments/data-generation.yaml]: yes! done

File: pipelines/azureml/pipelines/data_generation.py [hunk 4]
```diff
@@ -1,195 +1,161 @@
...
+    # you'll need a workspace object to connect
+    workspace = azureml_connect(config)
+
+    # run the pipeline function with the given arguments
+    pipeline_instance = data_generation_main_pipeline_function(config)
+
+    # generate a nice markdown description
+    experiment_description="\n".join([
+        "Generating synthetic datasets (see yaml below).",
+        "```yaml""",
```
[the remark this answers is not in the document]
Comment (jfomhover) [reply pipelines/azureml/pipelines/data_generation.py]: Thanks for spotting this, fixed.

**Pull request merged**